do_py/data_object/restriction.py

```diff
@@ -563,7 +564,7 @@ class Contact(DataObject):
 
     def __new__(cls, *args, **kwargs):
         cls._restriction = Restriction.legacy(cls._restriction)
-        return super(ManagedRestrictions, cls).__new__(cls, *args, **kwargs)
+        return super(ManagedRestrictions, cls).__new__(cls)
```

Comment:
Let's validate this change is well covered by unit tests and is regression free.